packages/holocron/src/holocronModule.jsx

```diff
@@ -64,25 +64,25 @@ export async function executeLoadingFunctions({
   // Frozen props as of when called
   frozenProps,
   // Provide loadCount to limit state changes
-  loadCount,
+  currentLoadCount,
   // Provide componentName for error messages
   componentName,
   // Provide component instance for getting current state of component post async
-  hocInstance,
+  hocInstance: { mounted, setState, loadCount },
 }) {
   try {
     // Call deprecated load function if exists
     await executeLoad(frozenProps);
     // Call loadModuleData if it exists
     await executeLoadModuleData(loadModuleData, WrappedComponent, frozenProps);
     // Modify state only when mounted and current loadCount is less or equal than previous loadCount
-    if (hocInstance.mounted && hocInstance.state.loadCount <= loadCount) {
-      hocInstance.setState({ status: 'loaded' });
+    if (mounted && loadCount <= currentLoadCount) {
+      setState({ status: 'loaded' });
```

Comment:
I think to keep the same functionality loadCount would need to be a ref. however i'm not sure that this logic still makes sense.

Reply:
Testing that with converting loadCount to ref. I don't see any reason that should be made state.